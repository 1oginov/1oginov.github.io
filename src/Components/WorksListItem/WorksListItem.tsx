import * as React from 'react';

import Work from 'Interfaces/Work';
import GitHubRepo from 'Lib/GitHub/GitHubRepo';
import { toGitHub, toNpm } from 'Lib/links';

interface Props {
  gitHubRepos?: GitHubRepo[];
  work: Work;
}

const WorksListItem: React.FunctionComponent<Props> = ({ work }: Props) => (
  <li>

    {work.archived ? <s>{work.name}</s> : work.name}

    {work.github && (
      <>
        {' '}
        <a href={toGitHub(work.name)}>GitHub</a>
      </>
    )}

    {work.npm && (
      <>
        {' '}
        <a href={toNpm(work.name)}>npm</a>
      </>
    )}

  </li>
);

export default WorksListItem;
