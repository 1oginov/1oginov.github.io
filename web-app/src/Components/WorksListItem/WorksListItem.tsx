import * as React from 'react';

import GitHubWorkDetails from 'Components/GitHubWorkDetails';
import Link from 'Components/Link';
import Work from 'Interfaces/Work';
import GitHubRepo from 'Lib/GitHub/GitHubRepo';
import { toGitHub, toNpm } from 'Lib/links';

interface Props {
  gitHubRepos?: GitHubRepo[];
  work: Work;
}

const WorksListItem: React.FunctionComponent<Props> = ({ gitHubRepos, work }: Props) => {
  const gitHubRepo = work.github && gitHubRepos ? gitHubRepos.find((repo) => repo.name === work.name) : undefined;

  let title = <strong>{work.name}</strong>;
  if (work.homepage) {
    title = <Link href={work.homepage}>{title}</Link>;
  }
  if (work.archived) {
    title = <s title="Archived">{title}</s>;
  }

  return (
    <li>

      {title}

      {work.github && (
        <>
          {' '}
          /
          {' '}
          <Link href={toGitHub(work.name)}>GitHub</Link>
        </>
      )}

      {work.npm && (
        <>
          {' '}
          /
          {' '}
          <Link href={toNpm(work.name)}>npm</Link>
        </>
      )}

      {work.platformio && (
        <>
          {' '}
          /
          {' '}
          <Link href={work.platformio}>PlatformIO</Link>
        </>
      )}

      {gitHubRepo && (
        <>
          <br />
          <GitHubWorkDetails repo={gitHubRepo} />
        </>
      )}

    </li>
  );
};

WorksListItem.defaultProps = {
  gitHubRepos: [],
};

export default WorksListItem;
