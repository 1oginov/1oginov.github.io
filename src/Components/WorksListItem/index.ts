import WorksListItem from './WorksListItem';

export default WorksListItem;
