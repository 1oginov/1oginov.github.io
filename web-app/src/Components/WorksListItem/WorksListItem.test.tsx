/* eslint-disable import/no-extraneous-dependencies */

import { shallow } from 'enzyme';
import * as React from 'react';

import mockData from 'Lib/GitHub/__fixtures__/data.json';

import { WorksListItem } from './WorksListItem';

it('matches snapshot', () => {
  const wrapper = shallow(<WorksListItem work={{ name: 'Some work' }} />);

  expect(wrapper).toMatchSnapshot();
});

it('matches snapshot with GitHub repo', () => {
  const wrapper = shallow(
    <WorksListItem
      gitHubRepos={mockData.repos}
      work={{ github: true, name: 'Angular-Gulp-Boilerplate' }}
    />,
  );

  expect(wrapper).toMatchSnapshot();
});
