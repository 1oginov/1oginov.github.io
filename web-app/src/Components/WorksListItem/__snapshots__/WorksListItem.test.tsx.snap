// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<li>
  <strong>
    Some work
  </strong>
</li>
`;

exports[`matches snapshot with GitHub repo 1`] = `
<li>
  <strong>
    Angular-Gulp-Boilerplate
  </strong>
   
  /
   
  <Link
    href="https://github.com/loginov-rocks/Angular-Gulp-Boilerplate"
  >
    GitHub
  </Link>
  <br />
  <GitHubWorkDetails
    repo={
      Object {
        "description": "Clean but full featured AngularJS boilerplate using Gulp workflow and best practices",
        "homepageUrl": "https://www.npmjs.com/package/angular-gulp-boilerplate",
        "language": "JavaScript",
        "latestVersion": "",
        "stars": 29,
        "title": "Angular-Gulp-Boilerplate",
        "updatedAt": 1613486363000,
        "url": "https://github.com/loginov-rocks/Angular-Gulp-Boilerplate",
      }
    }
  />
</li>
`;
